sylvia/src/into_response.rs

```diff
@@ -0,0 +1,90 @@
+use cosmwasm_std::{CosmosMsg, Empty, Response, StdError, StdResult, SubMsg};
+
+/// Trait converting `SubMsg` to one carrying another chain-custom message
+pub trait IntoMsg<C> {
+    fn into_msg(self) -> StdResult<SubMsg<C>>;
+}
+
+/// `SubMsg<Empty>` can be made into any `SubMsg<C>`
+impl<C> IntoMsg<C> for SubMsg<Empty> {
+    fn into_msg(self) -> StdResult<SubMsg<C>> {
+        let msg = match self.msg {
+            CosmosMsg::Wasm(wasm) => CosmosMsg::Wasm(wasm),
+            CosmosMsg::Bank(bank) => CosmosMsg::Bank(bank),
+            CosmosMsg::Staking(staking) => CosmosMsg::Staking(staking),
+            CosmosMsg::Distribution(distribution) => CosmosMsg::Distribution(distribution),
+            CosmosMsg::Custom(_) => Err(StdError::generic_err(
+                "Custom Empty message should not be sent",
+            ))?,
+            #[cfg(feature = "stargate")]
+            CosmosMsg::Ibc(ibc) => CosmosMsg::Ibc(ibc),
+            #[cfg(feature = "stargate")]
+            CosmosMsg::Stargate { type_url, value } => CosmosMsg::Stargate { type_url, value },
+            _ => Err(StdError::generic_err(format!(
+                "Unknown message variant: {:?}",
+                self
+            )))?,
+        };
+
+        Ok(SubMsg {
+            msg,
+            id: self.id,
+            gas_limit: self.gas_limit,
+            reply_on: self.reply_on,
+        })
+    }
+}
+
+pub trait IntoResponse<T> {
+    fn into_response(self) -> StdResult<Response<T>>;
+}
+
+impl<T> IntoResponse<T> for Response<Empty> {
+    fn into_response(self) -> StdResult<Response<T>> {
+        let messages = self
+            .messages
+            .into_iter()
+            .map(|msg| msg.into_msg())
+            .collect::<StdResult<Vec<_>>>()?;
```

Comment:
I personally prefer:
```rust
let messages: Vec<_> = self
  // ...
  .collect::<Result<_, _>>()?;
```

To minimize turbofish. However, this is minor nitpicking.